scrabble.js

```diff
@@ -1,16 +1,253 @@
-
 const Scrabble = {
-  score(word) {
-
+  score_per_char(){
+    return{
+      A:1, E:1, I:1, O:1, U:1, L:1, N:1, R:1, S:1, T:1, D:2, G:2
+      ,B:3, C:3, M:3, P:3
+      ,F:4, H:4, V:4, W:4, Y:4
+      ,K:5
+      ,J:8,X:8
+      ,Q:10,Z:10
+      }
   },
-  highestScoreFrom(arrayOfWords) {
 
-  },
+  score:function(word){
+    if (word === ''){
+        throw new Error('Does not allow an empty input.');
+    }
+    if (word.length > 7){
+        throw new Error('Does not allow words > 7 letters.');
+    }
+    if (word.length < 1){
+       throw new Error('Word length must be at least 1 character.')
+    }
+
+    if (/^[a-zA-Z]*$(?=)/.test(word) === false){
+      throw new Error('Word must only characters A - Z.')
+    }
+
+        let total = 0;
+        const bonus_points = 50;
+         if (word.length == 7){
+           total+= bonus_points;
+         }
+
+         for(let i = 0; i < word.length; i++){
+           total+= this.score_per_char()[word[i].toUpperCase()] }
+         return total
+          },
+  highestScoreFrom:function(arrayOfWords){
+      if (arrayOfWords.length === 0){
+        throw new Error('Pass in a minimum of one word.')
+      }
+      if(Array.isArray(arrayOfWords) === false){
+         throw new Error('Collection of words must be an array.')
+       }
+      let winner = arrayOfWords[0];
+      const that = this;
+      arrayOfWords.forEach(function(word){
+          if(that.score(word) > that.score(winner)){
+               winner = word
+          }else if (that.score(word) === that.score(winner)){
+              if(word.length === 7 ){
+                   winner = word
+              }else if((word.length < winner.length) && winner.length != 7){
+                   winner = word
+              }
+          }
+      });
+      return winner
+  }
 };
 
 Scrabble.Player = class {
+  constructor(name){
+    if (name === null){
+      throw new Error('Null is not a valid input for name.');
+    }
+    if (name === undefined){
+      throw new Error('Name must be assigned a value.');
+    }
 
-};
+    this.name = name;
+    this.plays = [];
+  }
 
+  play(word){
+    if (word === null){
```

Comment:
Since both `null` and `undefined` are falsy, you could just do

```javascript
if (!word) {
  throw new Error(`${word} is not valid word`);
}
```